src/components/Languages.vue

```diff
@@ -0,0 +1,57 @@
+<template>
+  <div>
+    <a
+      href="#"
+      class="chip"
+      v-for="(language, index) in languagesWithColors"
+      :key="index"
+      :style="{ backgroundColor: language.backgroundColor, color: language.textColor }"
+    >
+      {{ language.text }}
+    </a>
+  </div>
+</template>
+
+<script>
+import colors from "@/util/colors.json";
+import { hexToRgb, setContrast } from "@/util/colors";
+
+export default {
+  name: "Language",
+  props: {
+    languages: {
+      type: String,
+      required: true
+    }
+  },
+  computed: {
+    languagesWithColors() {
+      return this.languages.split(" ").map(language => {
+        if (colors[language]) {
```

Comment:
I think we can change this if-else block with:

```
const languageColor = colors[language] || "#e4e4e4";
return {
   text: language,
   backgroundColor: languageColor,
   textColor: setContrast(hexToRgb(languageColor))
};
```

I guess this produce the same result, what you think ?

Reply:
Not exactly, in this case you're forgetting about the default black `languageColor`, but you're right, this can be improved, and based on your suggestion this could do the trick.

```js
const backgroundColor = colors[language] || "#e4e4e4";
const textColor = colors[language] ? setContrast(hexToRgb(backgroundColor)) : "black";
return {
    text: language,
    backgroundColor,
    textColor
};
```